pkg/gohai/filesystem/filesystem.go

```diff
@@ -3,105 +3,75 @@
 // Copyright © 2015 Kentaro Kuribayashi <kentarok@gmail.com>
 // Copyright 2014-present Datadog, Inc.
 
-//go:build linux || darwin
-// +build linux darwin
-
 // Package filesystem returns information about available filesystems.
 package filesystem
 
 import (
-	"context"
 	"errors"
-	"fmt"
-	"os/exec"
-	"strings"
+	"strconv"
 	"time"
 )
 
-var dfCommand = "df"
-var dfOptions = []string{"-l", "-k"}
-var dfTimeout = 2 * time.Second
+type FileSystem struct{}
 
-func getFileSystemInfo() (interface{}, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), dfTimeout)
-	defer cancel()
+type MountInfo struct {
+	Name      string `json:"name"`
+	SizeKB    uint64 `json:"kb_size"`
+	MountedOn string `json:"mounted_on"`
+}
 
-	/* Grab filesystem data from df	*/
-	cmd := exec.CommandContext(ctx, dfCommand, dfOptions...)
+var timeout = 2 * time.Second
+var ErrTimeoutExceeded = errors.New("timeout exceeded")
 
-	// force output in the C locale (untranslated) so that we can recognize the headers
-	cmd.Env = []string{"LC_ALL=C"}
+const name = "filesystem"
 
-	out, execErr := cmd.Output()
-	var parseErr error
-	var result []interface{}
-	if out != nil {
-		result, parseErr = parseDfOutput(string(out))
-	}
-
-	// if we managed to get _any_ data, just use it, ignoring other errors
-	if len(result) != 0 {
-		return result, nil
-	}
-
-	// otherwise, prefer the parse error, as it is probably more detailed
-	err := execErr
-	if parseErr != nil {
-		err = parseErr
-	}
-	if err == nil {
-		err = errors.New("unknown error")
-	}
-	return nil, fmt.Errorf("df failed to collect filesystem data: %s", err)
+func (fs *FileSystem) Name() string {
+	return name
 }
 
-func parseDfOutput(out string) ([]interface{}, error) {
-	lines := strings.Split(out, "\n")
-	if len(lines) < 2 {
-		return nil, errors.New("no output")
-	}
-	var fileSystemInfo = make([]interface{}, 0, len(lines)-2)
-
-	// parse the header to find the offsets for each component we need
-	hdr := lines[0]
-	fieldErrFunc := func(field string) error {
-		return fmt.Errorf("could not find '%s' in `%s %s` output",
-			field, dfCommand, strings.Join(dfOptions, " "))
+func (fs *FileSystem) Collect() (interface{}, error) {
```

Comment:
The `Collector` interface used in `Gohai`'s main package means we have to keep returning an `interface{}`.

Unfortunately in order to not change the json output format the `Collect` function needs to stay because the `kb_size` is an integer and not a string like the payload expects (so we can't just `Marhshal` what `Get` returns)

Reply:
To be more precise, we could also reuse some of the tools of the new API but it would require handling arrays which is not really in the scope of this PR.
I wasn't even initially planning on implementing a new API here but did so because that work was started in the original PR.